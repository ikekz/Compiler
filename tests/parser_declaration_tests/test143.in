const
	c1 = 5;
var
	v1 : integer = c1;
type
	t1 = integer;
procedure p1;
begin
end;
function f1 : integer;
begin
end;
function f2 : double;
begin
end;
procedure p2;
begin
end;
type
	t2 = double;
var
	v2 : t2 = -3;
const
	c2 = -17.8;
begin
end.