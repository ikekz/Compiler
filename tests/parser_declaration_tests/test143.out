integer        type           integer
double         type           double
char           type           char
c1             const          integer        5
v1             var            integer        5
t1             type           integer
p1             procedure      args:
                              none
                              locals:
                              none
f1             function       return type: 
                              integer
                              args:
                              none
                              locals:
                              none
f2             function       return type: 
                              double
                              args:
                              none
                              locals:
                              none
p2             procedure      args:
                              none
                              locals:
                              none
t2             type           double
v2             var            double         -3.000000
c2             const          double         -17.800000
